filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
```

Comment:
Yes, this is basically what I'm saying. Besides, I wonder if we need a 'Setup/Init' method in Prospectorer or constructor method would to the trick. e.g.

    prospector, err = NewStdinProspector()
    ...
    prospector, err = NewLogProspector(p.registrar)

This would keep the Prospector interface minimal.